frontend/src/components/Login.jsx

```diff
@@ -1,89 +1,255 @@
+              <Col>
+                <h1 className="fw-bold text-center">Login</h1>
+              </Col>
+            </Row>
+            <Row className="px-4">
+              <Form noValidate onSubmit={handleSubmit}>
+                <Form.Group className="mb-4" controlId="formUserIdentifier">
+                  <Form.Label>Username or Email</Form.Label>
+                  <Form.Control
+                    className={
+                      identifierValidated && (!identifierFilled || loginError)
+                        ? "is-invalid"
+                        : ""
+                    }
+                    type="text"
+                    placeholder="Type your username or email"
+                    value={identifier}
+                    onChange={onIdentifierChange}
+                  />
+                  <Form.Control.Feedback type="invalid">
+                    {loginError
+                      ? ""
+                      : "Please provide a valid username or email."}
+                  </Form.Control.Feedback>
+                </Form.Group>
+                <Form.Group className="mb-1" controlId="formUserPassword">
+                  <Form.Label>Password</Form.Label>
+                  <InputGroup>
+                    <Form.Control
+                      className={
+                        passwordValidated && (!passwordFilled || loginError)
+                          ? "is-invalid"
+                          : ""
+                      }
+                      type={showPassword ? "text" : "password"}
+                      placeholder="Type your password"
+                      value={password}
+                      onChange={onPasswordChange}
                     />
-                    <span onClick={() => setShowPassword(!showPassword)}>
-                        {showPassword ? '👁️' : '🙈'}
-                    </span>
-                </div>
-
-                {passwordError && <p style={{color: 'red'}}>{passwordError}</p>}
-
-                <button type="submit">ENTRAR</button>
-                <a href="/signup">Clica aqui per registrar-te</a>
-            </form>
-        </div>
-    );
+                    <InputGroup.Text
+                      className="password-icon"
+                      onClick={() => setShowPassword(!showPassword)}
+                    >
+                      {showPassword ? <EyeSlash></EyeSlash> : <Eye></Eye>}
+                    </InputGroup.Text>
+                    <Form.Control.Feedback type="invalid">
+                      {loginError ? "" : "Please provide a valid password."}
+                    </Form.Control.Feedback>
+                  </InputGroup>
+                </Form.Group>
+                <Row className="mb-5">
+                  <Col sm={12}>
+                    <a
+                      href="/forgotpassword"
+                      className="text-decoration-none fs-6"
+                    >
+                      Forgot your password?
+                    </a>
```

Comment:
Consider using the "Link" component from "react-router-dom" instead of "a" to avoid a full page reload.

Reply:
Good point